File: spec/fixtures/ts_app/src/app/loan.ts
import type { Book } from "@domain/book";
import type { Repo } from "../domain/repo";
import { Service } from "../domain/service";

export class LoanRepo extends Service implements Repo {
  run(): void {}

  get(isbn: string): Book | undefined {
    if (isbn) {
      return { title: isbn };
    }
    return undefined;
  }
}

export function issue(isbn: string): string {
  if (!isbn) {
    return "";
  }
  return isbn;
}

function _audit(msg: string): string {
  return msg;
}

File: spec/fixtures/python_app/src/myapp/app/loan.py
import os
from myapp.domain.repo import Repo
from myapp.domain.models import Book


class LoanRepo(Repo):
    def get(self, isbn):
        if isbn:
            return isbn
        return None


def issue(repo, isbn):
    return repo.get(isbn)


def _audit(msg):
    return msg
